src/lib/operators/table_scan/column_is_null_table_scan_impl.cpp

```diff
@@ -26,36 +28,63 @@ std::string ColumnIsNullTableScanImpl::description() const {
   return "IsNullScan";
 }
 
-std::shared_ptr<RowIDPosList> ColumnIsNullTableScanImpl::scan_chunk(const ChunkID chunk_id) {
-  const auto& chunk = _in_table->get_chunk(chunk_id);
-  const auto& segment = chunk->get_segment(_column_id);
+void ColumnIsNullTableScanImpl::_scan_non_reference_segment(
+    const AbstractSegment& segment, const ChunkID chunk_id, RowIDPosList& matches,
+    const std::shared_ptr<const AbstractPosList>& position_filter) {
+  const auto data_type = segment.data_type();
 
-  auto matches = std::make_shared<RowIDPosList>();
+  resolve_data_type(data_type, [&](const auto data_type_t) {
+    using DataType = typename decltype(data_type_t)::type;
```

Comment:
It does not really change a lot, but I think using `resolve_segment_and_data_type` is a cleaner approach.

Reply:
`resolve_data_and_segment_type`. Then we can have a templated scan method that either does the scan on the AV (dictionary), only forwads to the NULL value vector scan (most other encodings), or does the extra checks and forwards (unencoded).

Reply:
As I stumpled over it, we cannot check for BaseDictionary when using the resolve method, right? We need to check for all dictionary segments manually?